api/graphql/resolvers/album.go

```diff
@@ -148,24 +150,44 @@ func (r *albumResolver) Thumbnail(ctx context.Context, obj *models.Album) (*mode
 
 	var media models.Media
 
-	err := r.Database.Raw(`
-		WITH recursive sub_albums AS (
-			SELECT * FROM albums AS root WHERE id = ?
-			UNION ALL
-			SELECT child.* FROM albums AS child JOIN sub_albums ON child.parent_album_id = sub_albums.id
-		)
-
-		SELECT * FROM media WHERE media.album_id IN (
-			SELECT id FROM sub_albums
-		) AND media.id IN (
-			SELECT media_id FROM media_urls WHERE media_urls.media_id = media.id
-		) LIMIT 1
-	`, obj.ID).Find(&media).Error
-
-	if err != nil {
-		return nil, err
+	fmt.Print(obj.CoverID)
+
+	if obj.CoverID == 0 {
+		if err := r.Database.Raw(`
+			WITH recursive sub_albums AS (
+				SELECT * FROM albums AS root WHERE id = ?
+				UNION ALL
+				SELECT child.* FROM albums AS child JOIN sub_albums ON child.parent_album_id = sub_albums.id
+			)
+
+			SELECT * FROM media WHERE media.album_id IN (
+				SELECT id FROM sub_albums
+			) AND media.id IN (
+				SELECT media_id FROM media_urls WHERE media_urls.media_id = media.id
+			) LIMIT 1
+		`, obj.ID).Find(&media).Error; err != nil {
+			return nil, err
+		}
+	} else {
+		if err := r.Database.Where("id = ?", obj.CoverID).Find(&media).Error; err != nil {
+			return nil, err
+		}
 	}
 
+	// err := r.Database.Raw(`
+	// 	WITH recursive sub_albums AS (
+	// 		SELECT * FROM albums AS root WHERE id = ?
+	// 		UNION ALL
+	// 		SELECT child.* FROM albums AS child JOIN sub_albums ON child.parent_album_id = sub_albums.id
+	// 	)
+	//
+	// 	SELECT * FROM media WHERE media.album_id IN (
+	// 		SELECT id FROM sub_albums
+	// 	) AND media.id IN (
+	// 		SELECT media_id FROM media_urls WHERE media_urls.media_id = media.id
+	// 	) LIMIT 1
+	// `, obj.ID).Find(&media).Error
+
```

Comment:
This should probably be deleted

Reply:
Good catch - I thought I'd removed all the commented out code, but apparently not!